temporal/server.go

```diff
@@ -202,7 +202,13 @@ func (s *Server) Start() error {
 		case primitives.FrontendService:
 			svc, err = frontend.NewService(params)
 		case primitives.HistoryService:
-			svc, err = history.NewService(params)
+			var historySvc *history.Service
+			histApp := fx.New(
+				fx.Supply(params),
+				history.Module,
+				fx.Populate(&historySvc))
```

Comment:
So for now, we are using fx to create history service instance. Eventually, we should move to use fx to drive the application lifecycles, e.g. do something like this in main.

```
app := fx.New(...)
app.Run()

```

Reply:
That is the further plan.
I want to make these smaller changes for now. Mostly because each change can cause questions on used approaches and it's easier to resolve those in smaller scale.